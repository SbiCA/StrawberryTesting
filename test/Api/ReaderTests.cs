using System.Threading.Tasks;
using FluentAssertions;
using Snapshooter.Xunit;
using Xunit;

namespace StrawberryTesting.Tests.Api
{
    public class ReaderTests : BaseApiTest
    {
        [Fact]
        public async Task GivenAnonymousUser_WhenGetBlogs_ReturnsPosts()
        {
            using var testServer = GivenTestHost();
            var blogClient = testServer.GetBlogClient();

            var operationResult = await blogClient.ReadBlogs.ExecuteAsync();
            
            operationResult.Should().MatchSnapshot();
        }
        
        [Fact]
        public async Task GivenReaderAndi_WhenGetBlogs_ReturnsPosts()
        {
            using var testServer = GivenTestHost();
            var blogClient = testServer.GetBlogClient(TestUsers.ReaderAndi);

            var operationResult = await blogClient.ReadBlogs.ExecuteAsync();
            
            operationResult.Should().MatchSnapshot();
        }
    }
}