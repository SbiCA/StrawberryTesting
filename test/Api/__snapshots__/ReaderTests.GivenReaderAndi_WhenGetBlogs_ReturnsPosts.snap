{
  "Data": {
    "Posts": [
      {
        "Title": "My first blog",
        "Published": "2020-08-01T00:00:00+02:00"
      }
    ]
  },
  "DataInfo": {
    "Posts": [
      {
        "__typename": "BlogPost",
        "Title": "My first blog",
        "Published": "2020-08-01T00:00:00+02:00"
      }
    ],
    "EntityIds": [],
    "Version": 1
  },
  "DataFactory": {},
  "Errors": [],
  "Extensions": {},
  "ContextData": {}
}
